### DISCUSSION TITLE
How does the navigation work when using instant loading?

### DISCUSSION THREAD

**Discussion by bradumbaugh** [squidfunk/mkdocs-material#2099] [Q&A] [answered]
Just saw that you've added an instant loading feature to make Material for MkDocs behave more like a single page app. So cool. 

I'm excited to give this a whirl, but quick question first: when I click a link in a doc's content, or an item in the left nav, does the left nav stay persistent (scroll position, nodes that are open, etc.)? Or does the left nav reload to its default state for the new destination?

Thanks!

**Comment by squidfunk** [comment, marked as answer]
Instant loading replaces designated parts of the site without reloading the whole page. The logic is rather "dumb", so it will just replace the sidebar HTML with the new HTML and rebind all downstream event listeners (implemented through RxJS observables), thus the scroll position is lost if the sidebar was scrolled before.

In theory, this could be mitigated with some extra JavaScript, but I haven't found the need to do so, yet.

**Reply by bradumbaugh** [reply]
OK, thanks, makes sense. Just for context, it can be a little disorienting on docs sites in general (not just Material for MkDocs) when the nav changes around (resets the scroll, changes open nodes, etc) when content loads. It's not a major issue though, was just curious how it works in this case.

Is this feature running on the Material for MkDocs site?

**Reply by squidfunk** [reply]
I agree that there's always room for improvement and content shift is always something I'm trying to minimize.

Instant loading is currently not activated in the official docs, but many people use it already.

**Reply by bradumbaugh** [reply]
Cool. Appreciate this tool, thanks for all your work on it!

**Comment by sebas-intellegens** [comment]
I ran into the same issue (I think)
Just to confirm, the sidebar jumping up (i.e. resetting/rerendering) is still expected behavior when scrolling down it and clicking on a link?
![Peek 2021-11-19 18-37](https://user-images.githubusercontent.com/55788126/142674967-3013134b-7259-4343-aa4f-54450aa77339.gif)

**Reply by squidfunk** [reply]
Yes. This is already solved in [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/) but not ready for release:

- #2155
